torch/distributed/device_mesh.py

```diff
@@ -199,7 +199,7 @@ def __init__(
         ) -> None:
             self.device_type = device_type
             self.mesh = (
-                mesh.detach()
+                mesh.detach().cpu()
```

Comment:
would it make sense to actually assert the passed mesh is on cpu? this way is probably fine, just i can barely imagine anyone passing a cuda tensor intentionally

Reply:
Yeah I'm ok with just throwing user error on it and let them pass CPU tensors instead